app/lib/rocketchat.js

```diff
@@ -508,10 +508,10 @@ const RocketChat = {
 		}
 		return Promise.resolve(result);
 	},
-	async getPermalink(message) {
+	async getPermalink(obj, type) {
 		let room;
 		try {
-			room = await RocketChat.getRoom(message.rid);
+			room = await RocketChat.getRoom(obj.rid);
```

Comment:
make sure `await RocketChat.getRoom` is necessary when the `type` is `'room'` and first argument `obj` is basically the room itself